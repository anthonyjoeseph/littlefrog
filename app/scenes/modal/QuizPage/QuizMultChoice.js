// app/QuizMultChoice.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableWithoutFeedback
} from 'react-native';

import Button from 'react-native-button'
import { RadioButtons } from 'react-native-radio-buttons';

const options = [
  "Option A",
  "Option B",
  "Option C",
  "Option D"
];

class QuizMultChoice extends Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedOptionIndex: -1
    }

    this.setSelectedOption = this.setSelectedOption.bind(this);
    this.renderOption = this.renderOption.bind(this);
    this.renderContainer = this.renderContainer.bind(this);
    this.answer = this.answer.bind(this);
  }

  setSelectedOption(selectedOption){
    var selectedOptionIndex = this.props.options.indexOf(selectedOption);
    this.setState({
      selectedOption,
      selectedOptionIndex: selectedOptionIndex
    });
  }

  renderOption(option, selected, onSelect, index){
    const style = selected ? { fontWeight: 'bold' } : {};

    return (
      <TouchableWithoutFeedback onPress={onSelect} key={index}>
        <View>
          <Text style={style}>{option}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }

  renderContainer(optionNodes){
    return <View>{optionNodes}</View>;
  }

  answer(){
    if(this.state.selectedOptionIndex == this.props.correctIndex){
      this.props.onCorrect();
    }else{
      this.props.onWrong();
    }
  }

  render() {
    return (
      <View>
        <Text style={{fontSize:25}}>{this.props.question}</Text>
        <RadioButtons
          options={ this.props.options }
          onSelection={ this.setSelectedOption }
          selectedOption={this.state.selectedOption }
          renderOption={ this.renderOption }
          renderContainer={ this.renderContainer }
        />
        <Text style={{color:'rgb(150, 150, 150)'}}>Selected option: {this.state.selectedOption || 'none'}</Text>
        <Button onPress={this.answer}>ANSWER</Button>
      </View>
    )
  }
}

export default QuizMultChoice;
