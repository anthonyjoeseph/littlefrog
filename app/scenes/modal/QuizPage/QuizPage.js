// app/QuizMultChoice.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableWithoutFeedback
} from 'react-native';

import { Actions } from 'react-native-router-flux';
import { RadioButtons } from 'react-native-radio-buttons';
import Button from 'react-native-button';

import DismissableFullscreenPopup from '../DismissableFullscreenPopup'
import QuizMultChoice from './QuizMultChoice'

const questionExpl = "What is my favorite food?";
const optionsExpl = [
  "Bananas",
  "Razors",
  "Fried Chicken",
  "Pencil Shavings"
];

class QuizPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      question: props.question,
      options: props.options,
      correctIndex: props.correctIndex,
      isShowingQuestion: true,
      isAnswerCorrect: false
    };

    this.popupContent = this.popupContent.bind(this);
  }

  popupContent(){
    if(this.state.isShowingQuestion){
      return (
        <QuizMultChoice
          question={this.state.question}
          options={this.state.options}
          correctIndex={this.state.correctIndex}
          onCorrect={() => { this.setState({isShowingQuestion: false, isAnswerCorrect: true}); } }
          onWrong={() => { this.setState({isShowingQuestion: false, isAnswerCorrect: false}); } }
        />
      );
    }else{
      return (
        <View style={{flexDirection:'column'}}>
          <Text>{(this.state.isAnswerCorrect ? 'Right!' : "Wrong!")}</Text>
          <Button onPress={()=>{this.refs.dismissableScreen.dismiss()}}>BACK</Button>
        </View>
      );
    }
  }

  render() {
    return (
      <DismissableFullscreenPopup
        ref='dismissableScreen'
        onDismiss={() => {Actions.pop();}}>
        <View style={styles.popupView}>
          {this.popupContent()}
        </View>
      </DismissableFullscreenPopup>
    )
  }
}

const styles = StyleSheet.create({
  popupView:{
    backgroundColor: 'white',
    width: 300,
    height: 200,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'stretch'
  }
});

export default QuizPage;
