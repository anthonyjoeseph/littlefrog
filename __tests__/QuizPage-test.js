// __tests__/QuizPage-test.js
import 'react-native';
import React from 'react';
import QuizPage from '../app/scenes/modal/QuizPage/QuizPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <QuizPage
      question="What is my favorite food?"
      options={[
        "Bananas",
        "Razors",
        "Fried Chicken",
        "Pencil Shavings"
      ]}
      correctIndex={2}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
