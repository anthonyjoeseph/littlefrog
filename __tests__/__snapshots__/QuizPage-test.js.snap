exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "flexDirection": "row",
      "height": 1334,
      "justifyContent": "center",
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": 750,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "flexDirection": "column",
            "height": 200,
            "justifyContent": "center",
            "width": 300,
          }
        }
        testID={undefined}>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 25,
              }
            }>
            What is my favorite food?
          </Text>
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={undefined}
              testID={undefined}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}>
                Bananas
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={undefined}
              testID={undefined}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}>
                Razors
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={undefined}
              testID={undefined}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}>
                Fried Chicken
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={undefined}
              testID={undefined}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={Object {}}>
                Pencil Shavings
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgb(150, 150, 150)",
              }
            }>
            Selected option: 
            none
          </Text>
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}>
            <View
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#0C42FD",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#841584",
                    },
                    Object {
                      "color": "#cdcdcd",
                    },
                  ]
                }>
                ANSWER
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
